pandas/core/describe.py

```diff
@@ -203,3 +123,97 @@ def describe_1d(data) -> "Series":
     d = concat([x.reindex(names, copy=False) for x in ldesc], axis=1, sort=False)
     d.columns = data.columns.copy()
     return d
+
+
+def describe_numeric_1d(series, percentiles) -> "Series":
+    from pandas import Series
+
+    formatted_percentiles = format_percentiles(percentiles)
+
+    stat_index = ["count", "mean", "std", "min"] + formatted_percentiles + ["max"]
+    d = (
+        [series.count(), series.mean(), series.std(), series.min()]
+        + series.quantile(percentiles).tolist()
+        + [series.max()]
+    )
+    return Series(d, index=stat_index, name=series.name)
+
+
+def describe_categorical_1d(data, is_series) -> "Series":
+    names = ["count", "unique"]
+    objcounts = data.value_counts()
+    count_unique = len(objcounts[objcounts != 0])
+    result = [data.count(), count_unique]
+    dtype = None
+    if result[1] > 0:
+        top, freq = objcounts.index[0], objcounts.iloc[0]
+        if is_datetime64_any_dtype(data.dtype):
+            if is_series:
+                stacklevel = 5
+            else:
+                stacklevel = 6
+            warnings.warn(
+                "Treating datetime data as categorical rather than numeric in "
+                "`.describe` is deprecated and will be removed in a future "
+                "version of pandas. Specify `datetime_is_numeric=True` to "
+                "silence this warning and adopt the future behavior now.",
+                FutureWarning,
+                stacklevel=stacklevel,
+            )
+            tz = data.dt.tz
+            asint = data.dropna().values.view("i8")
+            top = Timestamp(top)
+            if top.tzinfo is not None and tz is not None:
+                # Don't tz_localize(None) if key is already tz-aware
+                top = top.tz_convert(tz)
+            else:
+                top = top.tz_localize(tz)
+            names += ["top", "freq", "first", "last"]
+            result += [
+                top,
+                freq,
+                Timestamp(asint.min(), tz=tz),
+                Timestamp(asint.max(), tz=tz),
+            ]
+        else:
+            names += ["top", "freq"]
+            result += [top, freq]
+
+    # If the DataFrame is empty, set 'top' and 'freq' to None
+    # to maintain output shape consistency
+    else:
+        names += ["top", "freq"]
+        result += [np.nan, np.nan]
+        dtype = "object"
+
+    from pandas import Series
+
+    return Series(result, index=names, name=data.name, dtype=dtype)
+
+
+def describe_timestamp_1d(data, percentiles) -> "Series":
+    # GH-30164
+    from pandas import Series
```

Comment:
move imports

Reply:
Unfortunately for now it is necessary to keep the imports in these functions, otherwise there will be cyclic imports.